src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -89,40 +104,81 @@ const DotPlotPage = (props) => {
     if (hierarchy.length === 0) dispatch(loadCellSets(experimentId));
   }, []);
 
-  useEffect(() => {
-    if (config && config.selectedGenes.length === 0 && !genesFetching && !highestDispersionGenes) {
-      const PROPERTIES = ['dispersions'];
-      const tableState = {
-        pagination: {
-          current: 1, pageSize: config.nMarkerGenes, showSizeChanger: true, total: 0,
-        },
-        geneNamesFilter: null,
-        sorter: { field: PROPERTIES[0], columnKey: PROPERTIES[0], order: 'descend' },
-      };
+  const currentConfig = useRef(null);
+
+  const getCurrentDataProps = (conf) => ({
+    useCustomGenes: conf.useCustomGenes,
+    nMarkerGenes: conf.nMarkerGenes,
+    selectedGenes: conf.selectedGenes,
+    selectedCellSet: conf.selectedCellSet,
+    selectedPoints: conf.selectedPoints,
+  });
 
-      dispatch(loadPaginatedGeneProperties(experimentId, PROPERTIES, plotUuid, tableState));
+  const sameConfig = () => _.isEqual(
+    currentConfig.current,
+    getCurrentDataProps(config),
+  );
+
+  useEffect(() => {
+    if (config && !sameConfig()) {
```

Comment:
try to avoid these kind of functions `sameConfig()` where instead of passing parameters you rely on getting them from the class. It's harder to test and reason about these functions. If you have the function `sameConfig(cfg1, cfg2)` it is very easy to write a test for that. Instead, with `sameConfig()` how do you test it? you need to mock the class to get the `currentConfig.current` and then `config` is not even defined. 

Moreover you are using `getCurrentDataProps(config)` but `config` is not a parameter so you are relying on the variable having the same name in the caller function which is very brittle and will break if someone changes the name.

Reply:
I understand. I'm removing the wrapper function to remove the dependency.

`getCurrentDataProps` (renamed to `getPlotDataConfig`) actually takes and use a parameter - `conf`. It does not use the component's `config` object. I move the function out of the component body, so it's clearer.